external/graphdb-service.js

```diff
@@ -59,12 +60,12 @@ class GraphdbService {
 
             try {
                 const stream = await this.repository.query(payload);
-                const result = [];
+                const bl = new BufferList()
                 stream.on('data', (bindings) => {
-                    result.push(bindings);
+                    bl.append(bindings);
                 });
                 stream.on('end', () => {
-                    accept(result);
+                    accept(bl);
```

Comment:
Here we return BufferList, in construct method we return string and in ask method we return json. All three methods are querying the graphdb, so do we need all of them to return different data type or we can make it return the same, because maybe this inconsistency can give us some trouble in the future

Reply:
Good point, let's make it uniform